packages/node-config-provider/src/fromEnv.ts

```diff
@@ -0,0 +1,22 @@
+import { ProviderError } from "@aws-sdk/property-provider";
+import { Provider } from "@aws-sdk/types";
+
+export type GetterFromEnv<T> = (env: NodeJS.ProcessEnv) => T | undefined;
+
+/**
+ * Get config value given the environment variable name or getter from
+ * environment variable.
+ */
+export const fromEnv = <T = string>(envVarSelector: GetterFromEnv<T>): Provider<T> => async () => {
+  try {
+    const config = envVarSelector(process.env);
+    if (config === undefined) {
+      throw new Error();
```

Comment:
It gets rethrown on line 18. Doing this so that any errors thrown in `envVarSelector` will be rethrown as a `ProviderError`, similar to the `fromSharedConfigFiles.ts`.